csharp/ql/lib/semmle/code/csharp/frameworks/Razor.qll

```diff
@@ -0,0 +1,181 @@
+/** Provides definitions and flow steps related to Razor pages. */
+
+import csharp
+private import codeql.util.Unit
+private import codeql.util.FilePath
+private import semmle.code.csharp.frameworks.microsoft.AspNetCore
+
+/** A call to the `View` method */
+private class ViewCall extends MethodCall {
+  ViewCall() { this.getTarget().hasQualifiedName("Microsoft.AspNetCore.Mvc", "Controller", "View") }
+
+  /** Gets the `name` argument to this call, if any. */
+  string getNameArgument() {
+    exists(StringLiteral lit, int i | i in [0 .. 1] |
+      this.getTarget().getParameter(i).getType() instanceof StringType and
+      DataFlow::localExprFlow(lit, this.getArgument(i)) and
+      result = lit.getValue()
+    )
+  }
+
+  /** Gets the `model` argument to this call, if any. */
+  Expr getModelArgument() {
+    exists(int i | i in [0 .. 1] |
```

Comment:
There exist an overload of `View` where the model parameter is the third parameter - so maybe we should include `i = 2` in the range as well.

Reply:
I can't find an overload with three or more parameters in the documentation?

Reply:
Uh, I looked at the documentation for `[System.Web.Mvc](https://learn.microsoft.com/en-us/dotnet/api/system.web.mvc?view=aspnet-mvc-5.2)` - so please disregard.